src/EventRegistry.cc

```diff
@@ -172,11 +172,20 @@ void EventGroup::UpdateFuncBodies() {
     static auto is_group_disabled = [](const auto& g) { return g->IsDisabled(); };
 
     for ( auto& func : funcs ) {
-        for ( auto& b : func->bodies )
+        bool func_changed = false;
+        for ( auto& b : func->bodies ) {
+            auto prev = b.disabled;
             b.disabled = std::any_of(b.groups.cbegin(), b.groups.cend(), is_group_disabled);
+            if ( prev != b.disabled )
+                func_changed = true;
+        }
+
+        if ( ! func_changed )
+            continue;
```

Comment:
I'll rewrite this a bit to track the all some/all body state via `|=` and `&=` within the loop, and removing the `any_of` / `all_of` part.